modules/rtpengine/rtpengine_hash.h

```diff
@@ -0,0 +1,31 @@
+#ifndef _RTPENGINE_HASH_H
+#define _RTPENGINE_HASH_H
+
+#include "../../str.h"
+
+
+/* table entry */
+struct rtpengine_hash_entry {
+	unsigned int tout;			// call timeout
+	str callid;				// call callid
+	struct rtpp_node *node;			// call selected node
+
+	struct rtpengine_hash_entry *next;	// call next
+};
+
+/* table */
+struct rtpengine_hash_table {
+	struct rtpengine_hash_entry **entry_list;	// hastable
+	unsigned int total;				// total number of entries in the hashtable
+};
+
+
+int rtpengine_hash_table_init(int size);
+int rtpengine_hash_table_destroy();
+int rtpengine_hash_table_insert(void *key, void *value);
```

Comment:
I don't think there's a good reason to use void pointers here, since the types are known and using void pointers prevents the compiler from catching type errors.